packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -439,13 +434,11 @@ export default class FakeTimers<TimerRef> {
       uuid: String(uuid),
     });
 
-    const cancelledImmediates = this._cancelledImmediates;
+    const immediates = this._immediates;
     this._timerAPIs.setImmediate(() => {
-      if (!cancelledImmediates.hasOwnProperty(uuid)) {
-        // Callback may throw, so update the map prior calling.
-        cancelledImmediates[String(uuid)] = true;
+      this._fakeClearImmediate(String(uuid));
+      immediates.find(x => x.uuid === String(uuid)) &&
```

Comment:
Could you stick `String(uuid)` in a variable to avoid doing it for every iteration?